### PR TITLE
Add balance functionality

### PR DESCRIPTION
**Pull request by Gaff**
I'd like to propose adding functionality to easily check the total in / out / balance on an address. This is particularly useful for walking BIP32 trees in order to sweep an xprv or determine which is the next unused xpub derived address. As an added bonus it seems to tax the web-apis less than fetching the full transaction info for each address.

I've implemented this for biteasy and it seems OK. I may have a go at some of the other providers too. This API isn't final since fetching addresses one by one is rather slow and it would be best to provide a bulk method that can query multiple addresses at once.

Would you be happy to include this sort of functionality into the library? Do you have any preferences on what form exactly it should take?

Many thanks!

### CODE REVIEW

**Comment by richardkiss** [comment]
I ```git cherry-pick```ed the fix of your name.

This seems like it would be useful functionality in certain circumstances. In general, I am bearish on web APIs to the blockchain because they tend to requires clients to trust them without proof that could be easily provided using block headers and merkle trees. But I would be willing to merge this sort of thing.

File: pycoin/services/biteasy.py [hunk 1]
```diff
@@ -10,16 +10,54 @@
...
+                    "accept": "*/*",
+                    "User-Agent": "curl/7.29.0"})
+    d = urlopen(r).read()
+    json_response = json.loads(d.decode("utf8"))
+
+    return json_response
+
+
+def balance_for_address(bitcoin_address, confidence=1):
+    """
```
Comment (richardkiss) [inline pycoin/services/biteasy.py]: I think "confirmations" might be better than "confidence" since the notion of "confidence" is a bit squishy.

**Comment by Gaff** [comment]
Thanks. Will change it to 'confirmations' to be more scientific and iterate a little and see how I get on.

I'm complete with you on the web apis, but currently pycoin uses them heavily elsewhere. In particular all the unspent output logic. What's your long-term vision here? Some sort of standard provider API where you can drop in bitcoind or blockchain.info as appropriate? [Even then how do you know your RPC connection to bitcoind is not under a MITM attack?]. I'd love to see something like that, any idea what it would look like?

In the short term the web-apis are convenient, and the risk is mitigated since you have to opt-in to any web api use either explicitly or via env variables I don't think it's terrible for now.

**Comment by richardkiss** [comment]
Rebased and merged.

**Pull request closed without merging**